5_data_munge/src/munge_data_files.R

```diff
@@ -0,0 +1,158 @@
+             is.na(`ActivityTopDepthHeightMeasure/MeasureValue`)&
+             is.na(`ActivityBottomDepthHeightMeasure/MeasureValue`)) %>%
+    filter(ActivityStartDate > as.Date('1800-01-01') & ActivityStartDate < as.Date('2020-01-01'))
+  
+ 
+  # now reduce to daily measures
+  # first clean up obviously bad data
+  dat_reduced <- filter(dat, ResultMeasureValue > min_value) %>%
+    filter(grepl('deg C|deg F', `ResultMeasure/MeasureUnitCode`, ignore.case = TRUE)) %>%
+    mutate(ResultMeasureValue = ifelse(grepl('deg F', `ResultMeasure/MeasureUnitCode`, ignore.case = TRUE),
+                                       f_to_c(ResultMeasureValue), ResultMeasureValue)) %>%
+    mutate(`ResultMeasure/MeasureUnitCode` = 'deg C') %>%
+    filter(ResultMeasureValue < max_value)
+  
+  dat_daily <- group_by(dat_reduced, MonitoringLocationIdentifier, ActivityStartDate) %>%
+    summarize(temperature_mean_daily = mean(ResultMeasureValue),
+              activity_start_times = paste(`ActivityStartTime/Time`, collapse = ', '),
+              n_day = n())
+  
+  # save
+  data_file <- scipiper::as_data_file(out_ind)
+  saveRDS(dat_daily, data_file)
+  gd_put(out_ind, data_file)
+  
+}
+
+munge_nwis_uv <- function(in_ind, min_value, max_value, max_daily_range, out_ind) {
+  nwis_df <- readRDS(sc_retrieve(in_ind)) %>%
+    filter(!is.na(temp_value), !is.na(dateTime)) %>%
+    filter(grepl('A', cd_value, ignore.case = FALSE)) %>%
+    filter(temp_value > min_value,
+           temp_value < max_value) %>%
+    mutate(dateTime = as.Date(dateTime)) %>%
+    group_by(site_no, col_name, dateTime) %>%
+    summarize(temperature_mean_daily = round(mean(temp_value), 1), n_day = n())
+  
+  # save
+  data_file <- scipiper::as_data_file(out_ind)
+  saveRDS(nwis_df, data_file)
+  gd_put(out_ind, data_file)
+}
+
+munge_nwis_dv <- function(in_ind, min_value, max_value, max_daily_range, out_ind) {
+
+  dat <- readRDS(sc_retrieve(in_ind)) %>%
+    filter(grepl('A', cd_value, ignore.case = FALSE)) %>%
+    filter(temp_value > min_value,
+           temp_value < max_value)
+  
+  # save
+  data_file <- scipiper::as_data_file(out_ind)
+  saveRDS(dat, data_file)
+  gd_put(out_ind, data_file)
+  
+}
+
+combine_all_dat <- function(wqp_ind, nwis_dv_ind, nwis_uv_ind, out_ind) {
+  wqp <- readRDS(sc_retrieve(wqp_ind)) %>%
+    mutate(date = as.Date(ActivityStartDate), source = 'wqp') %>%
+    select(site_id = MonitoringLocationIdentifier, 
+           date,
+           temp_degC = temperature_mean_daily,
+           n_obs = n_day, source)
+  
+  nwis_dv <- readRDS(sc_retrieve(nwis_dv_ind)) %>%
+    mutate(date = as.Date(dateTime),
+           source = 'nwiw_dv') %>%
+    select(site_id = site_no, 
+           date, temp_degC = temp_value,
+           source)
+  
+  nwis_uv <- readRDS(sc_retrieve(nwis_uv_ind)) %>%
+    ungroup() %>%
+    mutate(date = as.Date(dateTime),
+           source = 'nwiw_uv') %>%
+    select(site_id = site_no, 
```

Comment:
What do you think about making these NWIS site_ids be `USGS-XXX` here so they mesh with WQP conventions early on in the munging?

Reply:
Good idea.